crates/sidevm/host-runtime/src/env.rs

```diff
@@ -628,6 +647,47 @@ impl<'a, 'b> env::OcallFuncs for FnEnvMut<'a, &'b mut EnvInner> {
             (self.inner.gas_to_breath(&mut self.store) * 100 / self.gas_per_breath) as u8
         })
     }
+
+    fn query_local_contract(&mut self, contract_id: [u8; 32], payload: Vec<u8>) -> Result<i32> {
+        let sem = self
+            .inner
+            .outgoing_query_guard
+            .clone()
+            .try_acquire_owned()
+            .or(Err(OcallError::ResourceLimited))?;
+        let (res_tx, res_rx) = tokio::sync::mpsc::channel::<Vec<u8>>(1);
+        let res_id = self.resources.push(Resource::ChannelRx(res_rx))?;
+        let (reply_tx, reply_rx) = oneshot::channel();
+        let request = OutgoingRequest::Query {
+            contract_id,
+            payload,
+            reply_tx,
+        };
+        let from = self.inner.id;
+        self.inner
+            .outgoing_request_tx
+            .try_send((from, request))
+            .or(Err(OcallError::IoError))?;
+        tokio::spawn(async move {
+            let _sem = sem;
+            let result = match reply_rx.await {
+                Ok(reply) => res_tx.send(reply).await,
+                Err(_) => {
+                    warn!(target: "sidevm", "Failed to receive query result");
+                    res_tx.send(Vec::new()).await
```

Comment:
I guess this should never happen, even if the ink fails, right? Just want to confirm it doesn't hide error.

Reply:
Yes, It should never happen.